### ISSUE TITLE
train a new model for deep sequencing with the Linux system

### ISSUE THREAD

**Issue by WeiweiBian** [google/deepvariant#308]
Hi all,
I have read the discussion about deep sequencing on issue #62. I have tried to modify three options, downsample_fraction, pileup_image_height, and vsc_min_fraction_snps for our deep sequencing data but it didn't work and output many false-positive calls. Here I want to train a new model for deep sequencing data with rare somatic mutation(MAF~1%) and there is some confusion.
1) There is a maximum threshold for pileup_height of 362, can I modify it?
2) There is only one training tutorial with Google cloud platform, is there any guideline for training with the Linux system?
3) Can Deepvariant be adapted to a somatic mutation caller?
Thanks a lot!
Best regards,
Weiwei

**Comment by MariaNattestad** [comment]
Hi @WeiweiBian 

In our discussion over email, I recommended that you use a somatic caller instead of trying to adapt DeepVariant to your needs, and I still think this is the best way to go.

To answer your questions:
1. No, this is a limit within Inception V3 that DeepVariant uses.
2. We don't have any other training tutorials for other systems.
3. We have done some exploratory work on a somatic variant caller, but it is not available yet, and it is meant for tumor-normal pairs with much lower coverage and higher allele frequency. It will not be possible to use this for your case.

I think you would be much better off using a somatic variant caller for your research.
If you want to take transforming DeepVariant into a somatic caller on as a research project, you are welcome to do so, since it's open source. But we unfortunately don't have the bandwidth to guide you very much, since we have other exciting improvements to DeepVariant that we are already working on.

Best of luck with your work!
Maria

**Comment by WeiweiBian** [comment]
Hi Maria,
Thanks a lot for your help!
[problematic logs with training.docx](https://github.com/google/deepvariant/files/4641018/problematic.logs.with.training.docx)

Now I am setting Google Cloud SDK in our Linux system, but whenever I run the following command, there is an error.
( time seq 0 $((N_SHARDS-1)) | \
  parallel --halt 2 --joblog "${LOG_DIR}/log" --res "${LOG_DIR}" \
    sudo nvidia-docker run \
      -v ${HOME}:${HOME} \
      google/deepvariant:"${BIN_VERSION}-gpu" \
      /opt/deepvariant/bin/make_examples \
      --mode training \
      --ref "${REF}" \
      --reads "${BAM_CHR1}" \
      --examples "${OUTPUT_DIR}/training_set.with_label.tfrecord@${N_SHARDS}.gz" \
      --truth_variants "${TRUTH_VCF}" \
      --confident_regions "${TRUTH_BED}" \
      --task {} \
      --regions "'chr1'" \
) >"${LOG_DIR}/training_set.with_label.make_examples.log" 2>&1
Since there are no GPUs in our server, I changed the "nvidia-docker" to "docker", and the "${BIN_VERSION}-gpu" to "${BIN_VERSION}", but it still didn't work with an error that 'The index file is older than the data file'.
The log is attached here and I wonder whether the training mode could work without a GPU machine. Thanks a lot!
Best regards,
Weiwei

**Comment by pichuan** [comment]
Hi @WeiweiBian ,
first of all, the "The index file is older than the data file" warning is usually something to do with your BAM file or VCF file. It's usually just a warning and wouldn't affect the run. You should still be able to run the training. Looking at your error, it seems to me that you might not using the correct REF file that has the `chr` prefix. On my side, I will check to confirm that our training case study actually works as intended.

But, specifically for your use case --
To re-iterate what @MariaNattestad has mentioned, the current DeepVariant setup is not designed for your use case, and our team won't really have the bandwidth to support a very different use case like yours right now. Please consider using another tool that is designed for your use case.

If you plan to use TensorFlow (which is what we built on) for your own use case and explore the possibilities, that is great! Based on my experience, training with just CPUs would really slow for the training step (model_train and model_eval steps), I have not even attempted that myself to a real training run. I do not recommend it.

I will close this issue given that your use case is not a feature that our team will have bandwidth to support officially. Thanks!

**Comment by pichuan** [comment]
Hi @WeiweiBian ,
I re-ran the steps of make_examples on : https://github.com/google/deepvariant/blob/r0.10/docs/deepvariant-training-case-study.md

Our instructions work (with both CPU and GPU version) for make_examples.
